### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by volleyballkickedme**
Refactor code that handles mechanism to ensure no duplicate NRICs

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -42,41 +40,38 @@ public class AddCommand extends Command {
             + PREFIX_EMAIL + "johnd@example.com "
             + PREFIX_NRIC + "S1231231D "
             + PREFIX_ADDRESS + "311, Clementi Ave 2, #02-25 "
-            + PREFIX_TRIAGE + "1 "
-            + PREFIX_TAG + "Pacemaker";
+            + PREFIX_TRIAGE + "1"
+            + PREFIX_TAG + "Pacemaker "
+            + PREFIX_TAG + "High Blood Pressure";
```
Comment (Dominic-Khoo) [inline src/main/java/seedu/address/logic/commands/AddCommand.java]: This shouldn't work in our current implementation right? I thought our tags can't have whitespaces.

**Review by Dominic-Khoo — COMMENTED** [review]
Can see review comment, but other than that looks good, might need to smoke test to ensure that functionality not affected.

File: src/main/java/seedu/address/logic/commands/AddCommand.java (hunk 1 above)
Comment (volleyballkickedme) [reply src/main/java/seedu/address/logic/commands/AddCommand.java]: Was a typo mb. resolved.

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -81,12 +82,13 @@ public EditCommand(Nric nric, EditPersonDescriptor editPersonDescriptor) {
...
         this.nric = nric;
         this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
+        this.predicate = new NricMatchesPredicate(nric.toString());
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Person> lastShownList = model.getAddressBook().getPersonList();
+        /*List<Person> lastShownList = model.getAddressBook().getPersonList();
```
Comment (itsnotsherm) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: redundant code should be removed and not left as comments

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 3]
```diff
@@ -30,15 +28,20 @@ public class DeleteCommand extends Command {
...
     public DeleteCommand(Nric targetNric) {
         this.targetNric = targetNric;
+        this.predicate = new NricMatchesPredicate(targetNric.toString());
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Person> lastShownList = model.getAddressBook().getPersonList();
+        /*List<Person> lastShownList = model.getAddressBook().getPersonList();
```
Comment (itsnotsherm) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java]: same here do not leave this

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 4]
```diff
@@ -69,6 +69,7 @@ public class EditCommand extends Command {
 
     private final Nric nric;
```
Comment (itsnotsherm) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:70]: if you have a NricMatchesPredicate variable, I think having an Nric variable might no longer be necessary

File: src/main/java/seedu/address/logic/commands/LogCommand.java [hunk 5]
```diff
@@ -50,7 +50,7 @@ public LogCommand(Nric targetNric, Log log) {
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        List<Person> lastShownList = model.getAddressBook().getPersonList();
+        /*List<Person> lastShownList = model.getAddressBook().getPersonList();
```
Comment (itsnotsherm) [inline src/main/java/seedu/address/logic/commands/LogCommand.java]: remove redundant code

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java [hunk 6]
```diff
@@ -34,6 +32,7 @@ public class RemarkCommand extends Command {
     public static final String MESSAGE_DELETE_REMARK_SUCCESS = "Removed remark from Person: %1$s";
     private final Nric nric;
     private final Remark remark;
+    private final NricMatchesPredicate predicate;
```
Comment (itsnotsherm) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java:36]: as mentioned before

**Review by itsnotsherm — APPROVED** [review]
LGTM, just a few nits

**Pull request merged**